### DISCUSSION TITLE
[项目IDEA] Web3 OSS

### DISCUSSION THREAD

**Discussion by coderprepares** [IntensiveCoLearning/Web3-URL#166] [Ideas and Project Teams]
基于 ethstorage 实现对象存储（OSS）
通过已学习内容已经可以通过 ethfs + w3link 实现对象存储的公有访问
项目希望进一步实现将存储设为私有 并通过签名访问

目前有一些疑点：
1、现在用测试网上传文件消耗的测试币，不知转换成正式币大概是什么成本？
2、是否上传之后就能永久访问，访问带宽会有瓶颈吗？
3、上传文件的合约如果不开放公开访问的入口，是否能通过合约的交互信息通过其他方式获取到文件呢？

**Comment by qizhou** [comment]
> 
> 目前有一些疑点： 1、现在用测试网上传文件消耗的测试币，不知转换成正式币大概是什么成本？ 2、是否上传之后就能永久访问，访问带宽会有瓶颈吗？ 3、上传文件的合约如果不开放公开访问的入口，是否能通过合约的交互信息通过其他方式获取到文件呢？
1. 现在测试代币消耗比较大，是因为staking as storage的缘故。用正式代币的话，我们期望是$100~$10 per GB的量级。
2. 是永久保存的。读取的带宽瓶颈看节点网络的数目，以及是否有专门的服务节点（类似infura）。
3. 上传文件的合约可以见这里：https://github.com/ethstorage/evm-large-storage。

**Comment by qzhodl** [comment]
关于第三个问题，上传到EthStorage的数据都是公开的，即使上传合约限制了访问，也可以直接在EthStorage的存储节点查询到（只是更加费事）。如果需要保密，需要在应用中使用比如加密的技术来保护，可以参考https://github.com/ethstorage/w3drive的实现
